ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -200,7 +200,15 @@ private[apiserver] final class ApiSubmissionService private[services] (
   )(implicit loggingContext: LoggingContext): Future[Seq[SubmissionResult]] =
     if (configuration.implicitPartyAllocation) {
       val partiesInTransaction =
-        transaction.nodes.valuesIterator.flatMap(_.informeesOfNode).toSeq.distinct
+        transaction.nodes.valuesIterator
+          .flatMap {
+            // TODO https://github.com/digital-asset/daml/issues/8020
```

Comment:
rollback nodes don’t have informees so this seems perfectly fine.

Reply:
My concern is more about the rolled back nodes. Do we need to take into account the actions that have been rolled back ? It is not clear to me and I prefer to think properly about that in a different PR.

Reply:
Perfectly fine to do it in a separate PR, I’ll leave comments on the individual instances here anyway so the thoughts I put into it don’t get lost:

Implicit party allocation here: We clearly also need to allocate parties in rolled back nodes.